packages/terra-clinical-item-collection/src/Item.jsx

```diff
@@ -0,0 +1,189 @@
+      onSelect(event, itemKey);
+    };
+
+    selectableProps.onKeyDown = (event) => {
+      if (event.nativeEvent.keyCode === Utils.KEYCODES.ENTER || event.nativeEvent.keyCode === Utils.KEYCODES.SPACE) {
+        onSelect(event, itemKey);
+      }
+    };
+  }
+
+  return selectableProps;
+}
+
+function createListItem(elements, itemKey, selectableProps, isSelected, listItemStyles) {
+  const { startAccessory, children, comment, endAccessory, hasStartAccessory, hasEndAccessory, ...listItemProps } = elements;
+  // alignStartAccessory={hasStartAccessory}
+  // alignEndAccessory={hasEndAccessory}
+
+  const listItemContent = (
+    <ItemView
+      startAccessory={startAccessory}
+      displays={React.Children.toArray(children)}
+      comment={comment}
+      endAccessory={endAccessory}
+      {...listItemStyles}
+    />
+  );
+
+  return (
+    <List.Item
+      key={itemKey}
+      content={listItemContent}
+      isSelected={isSelected}
+      {...selectableProps}
+      {...listItemProps}
+    />
+  );
+}
+
+function createTableCell(content, keyValue, contentType, accessoryAlignment) {
+  const cellClassNames = cx(
+    `content-${contentType}`,
+    { 'content-accessory-align-center': (contentType === 'accessory' && accessoryAlignment === 'alignCenter') },
+    { 'content-accessory-align-top': (contentType === 'accessory' && accessoryAlignment === 'alignTop') },
+  );
+
+  return (<Table.Cell content={content} key={keyValue} className={cellClassNames} />);
+}
+
+function createTableRow(elements, itemKey, selectableProps, isSelected, accessoryAlignment) {
+  const { startAccessory, children, comment, endAccessory, ...tableRowProps } = elements;
+
+// QUESTION:
+  // Does ITEM as a standalone component want to care about the maximun number of displays columns?
+  // The max is 8 - determined by ItemView rendering.
+  // When used in the ItemCollection, _createTableView handles the case of too many.
+
+  let displayContent = [];
+  displayContent = React.Children.map(children, (display, index) => {
+    const displayKey = `display_${index + 1}`;
+    return createTableCell(display, displayKey, 'display');
+  });
+
+  return (
+    <Table.Row isSelected={isSelected} {...selectableProps} {...tableRowProps} key={itemKey}>
+      {startAccessory && createTableCell(startAccessory, 'start_accessory', 'accessory', accessoryAlignment)}
+      {displayContent}
+      {comment && createTableCell(comment, 'comment', 'comment')}
+      {endAccessory && createTableCell(endAccessory, 'end_accessory', 'accessory', accessoryAlignment)}
+    </Table.Row>
+  );
+}
+
+const Item = (props) => {
+  const {
+    view,
+    isSelectable,
+    isSelected,
+    onSelect,
+    listItemLayout,
+    listItemTextEmphasis,
+    isListItemTruncated,
+    accessoryAlignment,
+    ...otherItemProps
+  } = props;
+  // The itemKey is set by the ListView & TableView components as the child.key value.
+  const { itemKey, ...elements } = otherItemProps;
```

Comment:
So... this seems questionable to me. You using custom props in order to send the key, I assume so that this prop is not visible to consumers. Then you throw an error if we don't have it, essentially making it a required prop, but it's not documented anywhere except the console error.

Reply:
So I 100% agree this seems very odd. React [deprecated the ability for a component to access its own key](https://reactjs.org/blog/2014/10/16/react-v0.12-rc1.html#breaking-change-key-and-ref-removed-from-thisprops) a while back. The response was to create a different 'read-only' prop to pass the key down to the component. 

The ItemCollection handles this mapping to the itemKey 'read-only' prop [here](https://github.com/cerner/terra-clinical/pull/221/files#diff-fce8e1aeaf3a85769a01bcfc4c3bd946R61) and [here](https://github.com/cerner/terra-clinical/pull/221/files#diff-a8fa6bc226c2f3aa668a0e94d6f34228R36). 

Because we'd expect people to be using keys to manager the children, I didn't expose the itemKey prop because a consumer wouldn't need to define the itemKey & key props. They would only care about itemKey if they used Item outside of the ItemCollection.

I could do a couple things:
1. Expose the ItemKey prop with documentation regarding it should only be used when Item is not placed in the ItemCollection
2. Expose the prop, honor itemKey over key

I feel option two could becoming more overhead than its worth. I do thinking option one might be okay since @StephenEsser mention people are looking to create list-only or table-only collection and this would give them them the structor for consistent styling.

I did include an error console message if key was not provided in createSelectableProps() to indicate if key wasn't provided. I found that React requires keys/unique keys, it will accept null key values. 

@cerner/terra thoughts on this?

Reply:
I like option 1, would there be any issues if providing this prop if used in item collection?

Reply:
If provided, it would be overwritten by the key value provided.

Reply:
I'm still ok with option 1 then

Reply:
I vote for 1 as well

Reply:
@jmsv6d What are your thoughts?

Reply:
I'm okay with option 1, but it wouldn't be my first choice. As is, I definitely think that documentation would be needed, especially if people will be wanting to use this component outside of the item collection. However, I think that some refactoring could be done to remove the need for it at all. It's really only needed as a parameter for the onSelect function. If that logic was lifted up to a parent component I don't think that you'd need the 'read-only' prop at all.

Reply:
hmmm okay. I see what you are saying. Good call. I made some changes in [this commit](https://github.com/cerner/terra-clinical/pull/221/commits/7d8429c76e241a9157085530f2459bb32d290a53). Let me know if this is what you were thinking.

This commit moves the onSelect event allocation to the Utils files while keeping the isSelectable, tabIndex and isSelected logic within the Item file. Then for creating a selectable Item Collection there are no changes in implementation, however, anyone wanting to use Item outside of the IC can still receive tabIndex, isSelected and selectable styles but would need to attach their own onClick and onKeyDown functions for full selectability.